### PR TITLE
Improvements for release documentation

### PR DESCRIPTION
**Pull request by CoderDake**
I found the release process a bit hard to follow, so I've spent some time trying to improve the readability and clarity of some steps.

I also ran into an issue with build_e2e where the dds directory needed  a pub get, so I've just added that step so we don't need to run into that in the future

This isn't extensive but I think it helps :D

### CODE REVIEW

File: packages/devtools_app/lib/src/framework/release_notes/README.md [hunk 1]
```diff
@@ -1,55 +1,67 @@
...
 ### Testing the release notes in DevTools
-In `release_notes.dart` flip the `debugTestReleaseNotes` flag to true. 
+- In `release_notes.dart` flip the `debugTestReleaseNotes` flag to true. 
 
-Then, from the main `devtools/` directory, run the following:
-```dart
-dart ./tool/build_e2e.dart
-```
+- from the main `devtools/` directory, run the following:
+    ```dart
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/framework/release_notes/README.md:53]: should this be shell instead of dart?

File: tool/README.md [hunk 2]
```diff
@@ -3,9 +3,16 @@
...
-2. `echo $LOCAL_DART_SDK` gives you path to the local checkout
-3. The local chackout is at `main` branch: `git rebase-update`.
+1. You have a local checkout of the Dart SDK
+   - (for getting started instructions, see [sdk/CONTRIBUTING.md](https://github.com/dart-lang/sdk/blob/main/CONTRIBUTING.md)).
+2. Ensure your .bashrc sets `$LOCAL_DART_SDK`
+    
+   ```shell 
+   DART_SDK_REPO_DIR=<Path to cloned dart sdk>
+   export LOCAL_DART_SDK=$DART_SDK_REPO_DIR/sdk
+   ```
```
Comment (kenzieschmoll) [inline tool/README.md:13]: won't this only set the env variable for this terminal? (i.e. not long lasting?)

File: tool/README.md [hunk 3]
```diff
@@ -71,93 +88,121 @@ From the git GUI tool or from github.com directly:
...
-```shell
-cd $LOCAL_DART_SDK
-git rebase-update
-third_party/devtools/update.sh <your commit hash like 8881a7caa9067471008a8e00750b161f53cdb843>
-```
+### Upload the DevTools binary to CIPD
+- `TARGET_COMMIT_HASH` should be the commit hash you want to release DevTools from
+   - this should match the
+commit hash for the tag you just created
+   - and the [update.sh](https://github.com/dart-lang/sdk/blob/master/third_party/devtools/update.sh)
```
Comment (kenzieschmoll) [inline tool/README.md:141]: Use the update.sh script to build and upload the DevTools binary to CIPD: (then remove where this is stated below).

File: tool/README.md [hunk 4]
```diff
@@ -71,93 +88,121 @@ From the git GUI tool or from github.com directly:
...
+
+- Verify that running `dart devtools` launches the version of DevTools you just released. 
+   - For non-OSX
+      ```shell
+      out/ReleaseX64/dart-sdk/bin/dart devtools
+      ```
+   - for OSX
+      ```shell
+      xcodebuild/ReleaseX64/dart-sdk/bin/dart devtools
+      ```
```
Comment (kenzieschmoll) [inline tool/README.md:184]: reverse the order of these two since we mostly develop on mac

File: tool/README.md [hunk 5]
```diff
@@ -71,93 +88,121 @@ From the git GUI tool or from github.com directly:
...
+
+- If the version of DevTools you just published to CIPD loads properly
```
Comment (kenzieschmoll) [inline tool/README.md:186]: make a note that you may need to clear your browser history or open in an incognito window to avoid issues with Chrome aggressively caching

File: tool/README.md [hunk 6]
```diff
@@ -71,93 +88,121 @@ From the git GUI tool or from github.com directly:
...
+   - push up the SDK CL for review.
+      ```shell
+      git add . && \
+      git commit -m "Bump DevTools DEP to 2.16.0" && \
```
Comment (kenzieschmoll) [inline tool/README.md:190]: use the $NEW_DEVTOOLS_VERSION var here

File: tool/README.md (hunk 2 above)
Comment (CoderDake) [reply tool/README.md:13]: The instruction right above is to set this in bashrc,  so the assumption is that those lines would be set there :)

**Review by kenzieschmoll — APPROVED** [review]
